src/main/resources/jupyter/safe-mode.js

```diff
@@ -14,88 +14,43 @@
 // the intended use of this plugin is in a separate space from normal operation
 // which does not support localization features.
 
-define([
-    'base/js/namespace'
-], (Jupyter) => {
-  const load = () => {
-    // TODO: query welder for when to enable (IA-979).   always-on, for now
-    const enabled = true;
-    if (!enabled) {
-      return;
-    }
-
-
-    //
-    // Disable UI controls
-    //
-
-
-    // Save Notebook button
-    //"notbook" is an intentional typo to match the Jupyter UI HTML.
-    $('#save-notbook').remove();
-
-    // New Notebook menu tree
-    $('#new_notebook').remove();
-
-    // Open... menu item
-    $('#open_notebook').remove();
-
-    // Make a Copy... menu item
-    $('#copy_notebook').remove();
-
-    // Save as... menu item
-    $('#save_notebook_as').remove();
-
-    // Save and Checkpoint menu item
-    $('#save_checkpoint').remove();
-
-    // Revert to Checkpoint menu tree
-    $('#restore_checkpoint').remove();
-
-    // A little cleanup: remove two dividers in the file menu
-    // which are no longer dividing anything
-    $("#file_menu .divider")[0].remove();
-    $("#file_menu .divider")[0].remove();
-
-
-    //
-    // Disable UI notifications
-    //
+// const namespace = require('base/js/namespace')
 
+define(() => {
+    function load() {
+        checkMetaLoop()
+    }
 
-    $('#save_widget')
-        .append(
-            '<style>' +
-              // e.g. Last Checkpoint: 3 minutes ago
-              '.checkpoint_status { display: none; } ' +
-              // e.g. (autosaved) or (unsaved changes)
-              '.autosave_status { display: none; }' +
-              '</style>');
+    function checkMetaLoop() {
+        toggleUIControls(isSafeMode())
 
-    $('#notification_area')
-        .append(
-            '<style>' +
-              // e.g. Notebook Saved
-              '#notification_notebook span { display: none; } ' +
-              '</style>');
+        const interval = setInterval(() => {
```

Comment:
Is this going to change during the life of a notebook?  If not, we don't need to poll.

When we transition between safe mode and other modes, we will be copying the notebook file to a new directory.  Will that re-initialize this plugin, or will the current instance survive that transition?

Reply:
I think it may be worth polling to design for the possibility of switching in the future. 

As its currently envisioned,  when a user wishes to transfer from playground mode to edit mode or vice versa it will result in notebook reload and thus the plugin will re-trigger, so polling is not needed  as is.

I'm open to removing it for now, but I think the performance overhead would be negligible.

Reply:
ok to keep

Reply:
In Jupyter, it should not be possible to change notebooks within a single client-side session. Technically a notebook could change by way of /storageLinks, but this should never happen and would probably indicate deeper bugs if it did.

I would just lock in whatever mode you get back from the first checkMetadata request.